### DISCUSSION TITLE
All my photos have disappeared from PhotoPrism except one

### DISCUSSION THREAD

**Discussion by jean-louis67** [photoprism/photoprism#4038] [Q&A]
I have been using PhotoPrism for almost five years with great pleasure!
Yesterday afternoon I commented as usual on new photos using exiftool and launching the indexing
At one point, there was only one photo left even though I have around 12,000:
![image](https://github.com/photoprism/photoprism/assets/65403912/9fbda02d-42dc-44a7-aca6-dd8b3b8bcf65)
The photos are still present in the original directory. But the PhotoPrism application seems not to see them:
Example from directory 2024/01:
![image](https://github.com/photoprism/photoprism/assets/65403912/09ba5a9d-c6a8-4c14-b259-676ed29f6442)

I reindexed this directory like this, without result:
![image](https://github.com/photoprism/photoprism/assets/65403912/4b92d774-8e32-4d7b-8f67-6ce7a25d2c61)

I checked the database which seems correct, and if in doubt, I restored the backup from the previous night, but everything seems good on this side.
![image](https://github.com/photoprism/photoprism/assets/65403912/edbf439d-1c80-4a0d-bfaa-0beede1f92c8)

I emptied the storage/cache directory, without success

PhotoPrism was installed with docker-compose.
The files are installed on a file server and mounted like this: `/media/photoprism/originals:/photoprism/originals`
On the file server side (`/media/photoprism/originals/2024/01), the files are present and are not corrupted, the images displaying correctly in a file explorer.

I connected to the container with docker exec, and checked that the files were clearly visible:
![image](https://github.com/photoprism/photoprism/assets/65403912/1c07666b-09b5-49fc-b744-abe24b35f149)

I recreated the PhotoPrism and MariaDB containers, reloading the metadata, without further success.

I have nothing against reindexing the entire database, but it is very long and as it did not have the expected result on a directory, I am not sure that it would be of any use.

In short, I don't really know what to do, apart from looking at my last photo which reminds me of my vacation in Freiburg-in-Breisgau to improve my German... in 1970! :smiley: 
A little help would be welcome and I thank you in advance.
Jean-Louis

**Comment by graciousgrey** [comment]
I am happy to help!

Could you check the logs while indexing a subfolder, those might give some hints.

https://docs.photoprism.app/getting-started/troubleshooting/logs/#__tabbed_1_3

You might also want to go through this checklist:
https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

**Reply by jean-louis67** [reply]
Thank you very much for your answer.

* I now have a second photo, because there was an archived photo!
* I don't have any recent errors in Library->Errors, except for some _thumbs not found_
* Indexing logs: the 10 images are found and processed, but ultimately, they still do not appear in the PhotoPrism search or in the library. I don't really see in the log what could explain this.

Here are the logs for the indexing, that I did at least twice this morning:

`time="2024-01-29T06:36:20Z" level=info msg="indexation des fichiers de 2024/01"
time="2024-01-29T06:36:20Z" level=info msg="index: found no .ppignore file"
time="2024-01-29T06:36:23Z" level=info msg="media: created 11 thumbnails for 2024/01/20240106_154603.jpg [1.906985836s]"
time="2024-01-29T06:36:23Z" level=info msg="index: updated main jpg file 2024/01/20240106_154603.jpg"
time="2024-01-29T06:36:25Z" level=info msg="media: created 11 thumbnails for 2024/01/20240106_170137.jpg [2.330894347s]"
time="2024-01-29T06:36:26Z" level=info msg="index: updated main jpg file 2024/01/20240106_170137.jpg"
time="2024-01-29T06:36:27Z" level=info msg="media: created 11 thumbnails for 2024/01/20240110_152741.jpg [1.917355197s]"
time="2024-01-29T06:36:28Z" level=info msg="index: updated main jpg file 2024/01/20240110_152741.jpg"
time="2024-01-29T06:36:30Z" level=info msg="media: created 11 thumbnails for 2024/01/20240114_121031.jpg [1.797149728s]"
time="2024-01-29T06:36:30Z" level=info msg="index: updated main jpg file 2024/01/20240114_121031.jpg"
time="2024-01-29T06:36:32Z" level=info msg="media: created 11 thumbnails for 2024/01/20240114_133113.jpg [1.859124642s]"
time="2024-01-29T06:36:32Z" level=info msg="index: updated main jpg file 2024/01/20240114_133113.jpg"
time="2024-01-29T06:36:34Z" level=info msg="media: created 11 thumbnails for 2024/01/20240120_141032.jpg [1.921595075s]"
time="2024-01-29T06:36:34Z" level=info msg="index: updated main jpg file 2024/01/20240120_141032.jpg"
time="2024-01-29T06:36:36Z" level=info msg="media: created 11 thumbnails for 2024/01/20240121_100233.jpg [1.735427544s]"
time="2024-01-29T06:36:37Z" level=info msg="index: updated main jpg file 2024/01/20240121_100233.jpg"
time="2024-01-29T06:36:38Z" level=info msg="media: created 11 thumbnails for 2024/01/20240121_100417.jpg [1.633199362s]"
time="2024-01-29T06:36:38Z" level=info msg="index: updated main jpg file 2024/01/20240121_100417.jpg"
time="2024-01-29T06:36:39Z" level=info msg="media: created 10 thumbnails for 2024/01/Resized_IMG-20240104-WA0016.jpg [448.626153ms]"
time="2024-01-29T06:36:39Z" level=info msg="index: updated main jpg file 2024/01/Resized_IMG-20240104-WA0016.jpg"
time="2024-01-29T06:36:40Z" level=info msg="media: created 9 thumbnails for 2024/01/Resized_miteneB5A96A18-EE88-44E5-89FA-55D7DBBF6450.jpg [314.139144ms]"
time="2024-01-29T06:36:40Z" level=info msg="index: updated main jpg file 2024/01/Resized_miteneB5A96A18-EE88-44E5-89FA-55D7DBBF6450.jpg"
time="2024-01-29T06:37:10Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T06:37:10Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T06:37:10Z" level=info msg="index: updated 10 files [49.95420552s]"
time="2024-01-29T06:37:11Z" level=info msg="indexation terminée en 50 s"
time="2024-01-29T06:40:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T06:40:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T06:55:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T06:55:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T07:10:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T07:10:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T07:25:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T07:25:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T07:40:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T07:40:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T07:55:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T07:55:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T08:10:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T08:10:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T08:25:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T08:25:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T08:40:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T08:40:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T08:55:21Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T08:55:21Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T09:10:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T09:10:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T09:25:21Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T09:25:21Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T09:40:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T09:40:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T09:55:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T09:55:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T10:10:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T10:10:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T10:25:19Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T10:25:19Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T10:25:48Z" level=info msg="indexation des fichiers de 2024/01"
time="2024-01-29T10:25:48Z" level=info msg="index: found no .ppignore file"
time="2024-01-29T10:25:49Z" level=info msg="index: updated main jpg file 2024/01/20240106_154603.jpg"
time="2024-01-29T10:25:50Z" level=info msg="index: updated main jpg file 2024/01/20240106_170137.jpg"
time="2024-01-29T10:25:50Z" level=info msg="index: updated main jpg file 2024/01/20240110_152741.jpg"
time="2024-01-29T10:25:50Z" level=info msg="index: updated main jpg file 2024/01/20240114_121031.jpg"
time="2024-01-29T10:25:51Z" level=info msg="index: updated main jpg file 2024/01/20240114_133113.jpg"
time="2024-01-29T10:25:51Z" level=info msg="index: updated main jpg file 2024/01/20240120_141032.jpg"
time="2024-01-29T10:25:51Z" level=info msg="index: updated main jpg file 2024/01/20240121_100233.jpg"
time="2024-01-29T10:25:52Z" level=info msg="index: updated main jpg file 2024/01/20240121_100417.jpg"
time="2024-01-29T10:25:52Z" level=info msg="index: updated main jpg file 2024/01/Resized_IMG-20240104-WA0016.jpg"
time="2024-01-29T10:25:52Z" level=info msg="index: updated main jpg file 2024/01/Resized_miteneB5A96A18-EE88-44E5-89FA-55D7DBBF6450.jpg"
time="2024-01-29T10:26:21Z" level=info msg="faces: found 1 new cluster"
time="2024-01-29T10:26:21Z" level=info msg="faces: skipped cluster 5VCLSB7P2VB7TYWCDO4PCET5X6LAR7OB, embedding not distinct enough"
time="2024-01-29T10:26:21Z" level=info msg="index: updated 10 files [33.554596793s]"
time="2024-01-29T10:26:22Z" level=info msg="indexation terminée en 34 s"`

**Reply by graciousgrey** [reply]
Do you see photos in the review section? Is it possible that those files got stacked?

**Reply by jean-louis67** [reply]
Thank you very much for your suggestions.

I don't have a Review menu, perhaps because there are no photos to review. 

But, for my part, I found by looking in the database that absolutely all `photo_path` and `photo_name` have been overwritten! I don't understand at all how this happened. And that's probably the cause of the problem. Yesterday I added this Hengwiller/Histoire directory to the originals with only 18 photos, and now the paths of 12,000 photos have been changed!
![image](https://github.com/photoprism/photoprism/assets/65403912/e0201c1d-8c8c-4bc7-8346-3e4b3038352a)

 I then saw that the restoration I had done did not work, though I put `--force`:
`+ docker-compose -f /media/s-docker1/docker-compose/photoprism.yml exec -T photoprism photoprism restore -i /var/lib/photoprism/mysql/photoprism_20240128-003306.sql --force
time="2024-01-29T13:58:55Z" level=info msg="config: case-insensitive file system detected"
time="2024-01-29T13:58:55Z" level=error msg="found exisisting index with 12681 pictures, use --force to replace it"`
I have to find the right syntax, or start from an empty base. In any case, I now have good hope of getting through it!
I'll come back to post as soon as I succeed!
Thanks again for trying to help me.

**Reply by jean-louis67** [reply]
Ouf! (en:Phew!, de:Puh!)
I removed the database files `/root/photoprism/database` connected  to `/var/lib/mysql`, recreated the containers for PhotoPrism and MariaDB. 
The restoration then finally worked… and everything was back to normal! I don't understand what happened, but the main thing is to restore the situation.
Thank you again for your interest in my problem and your help.
Jean-Louis

**Reply by graciousgrey** [reply]
Glad to hear that it is working now :)
